stan/math/rev/functor/reduce_sum.hpp

```diff
@@ -216,12 +269,14 @@ struct reduce_sum_impl<ReduceFunction, require_var_t<ReturnType>, ReturnType,
     save_varis(varis, vmapped);
     save_varis(varis + num_vars_sliced_terms, args...);
 
-    for (size_t i = 0; i < num_vars_sliced_terms; ++i) {
-      partials[i] = 0.0;
+    Eigen::VectorXd shared_terms_adjoints(num_vars_shared_terms);
+
+    for (size_t i = 0; i < num_vars_shared_terms; ++i) {
+      shared_terms_adjoints(i) = varis[num_vars_sliced_terms + i]->adj_;
```

Comment:
`shared_terms_adjoints.coeffRef(i)` avoids out of bounds checks that `operator()` does for eigen types.

`shared_terms_adjoints.coeff(i)` is the version to use for reading (const).

Reply:
Ok. Will change.